src/trix/models/composition.coffee

```diff
@@ -298,6 +314,14 @@ class Trix.Composition extends Trix.BasicObject
   canDecreaseBlockAttributeLevel: ->
     @getBlock()?.getAttributeLevel() > 0
 
+  canAddBlockAttribute: (attributeName) ->
+    return true unless @getBlock()?.hasAttributes()
+    if attributeName in ["quote", "code", "bullet", "number"]
+      if "header" in @getBlock()?.getAttributes() and attributeName not in @getBlock()?.getAttributes()
```

Comment:
Explicitly listing all other block attributes here isn't going to be maintainable. Can this be extracted as a generalized configuration of a block in `Trix.config.blockAttributes`?